### PR TITLE
95-Use attributes in kubernetes component to store outerloop deployment config

### PR DESCRIPTION
**Pull request by yangcao77**
For issue:https://issues.redhat.com/browse/DEVHAS-95

The current behavior is injecting all HAS configs to container attributes, and some uses container endpoint (i.e. TargetPort). some uses container ENV. This behavior does not make sense, since the configs are for outerloop builds. 

This PR moves all those settings to attributes under the `kubernetes` component which is used by outer loop builds.
The PR also change the prefix of the attributes to more generic one, `deployment/<setting>`, which can be easily adopted by other tools. 

example of a HAS component with kubernetes attribute:
```
- attributes:
    deployment/containerENV:
    - name: FOO
      value: foo1
    - name: BAR
      value: bar1
    deployment/containerimage-port: 1111
    deployment/cpuLimit: "2"
    deployment/cpuRequest: 700m
    deployment/memoryLimit: 500Mi
    deployment/memoryRequest: 400Mi
    deployment/replicas: 1
    deployment/route: route111
    deployment/storageLimit: 400Mi
    deployment/storageRequest: 200Mi
  kubernetes:
    uri: outerloop-deploy.yaml
  name: outerloop-deploy
```
To view full exmaple of the component 
[example-component-spec.txt](https://github.com/redhat-appstudio/application-service/files/8673108/example-component-spec.txt)

### CODE REVIEW

File: pkg/devfile/devfile.go [hunk 1]
```diff
@@ -111,13 +111,16 @@ func ConvertImageComponentToDevfile(comp appstudiov1alpha1.Component) (data.Devf
...
 	// Generate a stub container component for the devfile
+	// ToDo: devfile library should provide a kubernetes component writer to allow writing the inlined deployment spec
 	components := []v1alpha2.Component{
 		{
-			Name: "container",
+			Name: "kubernetes",
 			ComponentUnion: v1alpha2.ComponentUnion{
-				Container: &v1alpha2.ContainerComponent{
-					Container: v1alpha2.Container{
-						Image: comp.Spec.ContainerImage,
```
[the remark this answers is not in the document]
Comment (yangcao77) [reply pkg/devfile/devfile.go:120]: do we need to store the image for now? HAC is using the component spec. We can store it after we have the parser&writer for kubernetes component.

File: controllers/component_controller_test.go [hunk 2]
```diff
@@ -1608,99 +1604,80 @@ func verifyHASComponentUpdates(devfile data.DevfileData, checklist updateCheckli
...
 			} else if err != nil {
 				goPkgTest.Error(err)
-			} else if component.Container.CpuRequest != cpuRequestChecklist.String() {
-				goPkgTest.Errorf("expected: %v, got: %v", cpuRequestChecklist.String(), component.Container.CpuRequest)
+			} else if cpuRequest.(string) != cpuRequestChecklist.String() {
+				goPkgTest.Errorf("expected: %v, got: %v", cpuRequestChecklist.String(), cpuRequest)
 			}
 		}
 
-		// Check for container endpoint only for the first container
```
[the remark this answers is not in the document]
Comment (yangcao77) [reply controllers/component_controller_test.go:1687]: aren't the attributes being added to all Kubernetes components? (previously was added to all container components)
talking about that, should we limit each component can only have one kubernetes comp?

[the remark this answers is not in the document]
Comment (maysunfaisal) [reply controllers/component_controller_test.go:1687]: No i dont think we should have limits on Kubernetes components..
And I thought we were updating only 1 devfile component but we're not. Lets also jsut bring this up in the team meeting

[the remark this answers is not in the document]
Comment (maysunfaisal) [reply controllers/component_controller_test.go:1687]: I think my preferred approach would be to add it to one component and how do we find that one component? We check for the deploy command and see what component it references. But if we dont have a deploy command, then we can perhaps pick the 1st Kubernetes component if there are more than 1 Kubernetes component

[the remark this answers is not in the document]
Comment (yangcao77) [reply controllers/component_controller_test.go:1687]: If there is no apply command associate with a Kube component, then the kube component will never be reached, so it's a useless content in devfile. If you want to check the deploy command, what is the point of having a kube component without a deploy command? and also, what if the devfile contains multiple deploy command? just use the default one?

[the remark this answers is not in the document]
Comment (yangcao77) [reply controllers/component_controller_test.go:1687]: the current logic is adding the attributes to all container components, and if no container component, create one. https://github.com/redhat-appstudio/application-service/pull/112#discussion_r876191571 this is to create the container component

**Review by maysunfaisal — APPROVED** [review]

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**